Tweak the errors doc and add ApolloResponse.dataAssertNoErrors (#3534)
apollographql/apollo-android: docs/source/essentials/01-errors.mdx

}
```

## Ignoring partial data

> If you don't want to handle partial responses, you can simplify your error handling logic.`ApolloResponse.dataOrThrow` will return a non-nullable `data` and throws if there are any errors so that you can handle all eroors in a single `catch {}` block. :

Reviewer: ```suggestion
If you don't want to handle partial responses, you can simplify your error handling logic. `ApolloResponse.dataAssertNoErrors` will return a non-nullable `data` or throw if there are any errors so that you can handle all errors in a single `catch {}` block:
```

Author: Spelling mistakes aside, this was made on purpose so that we can merge this PR before we release beta04. The `dataAssertNoErrors` documentation changes are done in https://github.com/apollographql/apollo-android/pull/3510/commits/8a66bbdbaec3fac64c5a6c7864d1895a53147ea2

Reviewer: Oh I see! Makes sense. Yeah having the doc published before the new versions is tricky! 

But other instances of `dataOrThrow` were updated as part of this PR 🤔
